src/app/functions/server/breachResolution.ts

```diff
@@ -0,0 +1,245 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getL10n } from "./l10n";
+import AppConstants from "../../../appConstants.js";
+
+/**
+ * Equivalent of Typescript "enum"
+ * These enum types map to HIBP's breach data types, defined in HIBP's API
+ * Always reference enum instead of strings to avoid spelling error / typos (ie. BreachDataTypes.Passwords)
+ */
+const BreachDataTypes = {
```

Comment:
You could annotate this to conform to the type definition in `breaches.d.ts`, to ensure that we don't forget to update this when we update that.

Alternatively, instead of having that type there, you could replace line 27 with

```ts
} as const;
```

after which you could turn it into a type using:

```ts
export type BreachDataTypesType = typeof BreachDataTypes;
```

(And possibly come up with better names :sweat_smile: )

Reply:
That’s a good idea — done in: [5ed06c7](https://github.com/mozilla/blurts-server/pull/3085/commits/5ed06c772ae3772ddd07cbf52bc5ac936c1779d8).